src/Fixer/Whitespace/BlankLineBeforeControlStatementFixer.php

```diff
@@ -0,0 +1,255 @@
+{
+    /**
+     * @var array
+     */
+    private static $defaultConfiguration = array(
+        'return',
+    );
+
+    /**
+     * @var array
+     */
+    private static $tokenMap = array(
+        'break' => T_BREAK,
+        'continue' => T_CONTINUE,
+        'if' => T_IF,
+        'return' => T_RETURN,
+        'switch' => T_SWITCH,
+        'throw' => T_THROW,
+        'try' => T_TRY,
+    );
+
+    /**
+     * @var array
+     */
+    private $fixTokenMap = array();
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configure(array $configuration = null)
+    {
+        if (null === $configuration) {
+            $configuration = self::$defaultConfiguration;
+        }
+
+        foreach ($configuration as $controlStatement) {
+            if (!is_string($controlStatement) || !array_key_exists($controlStatement, self::$tokenMap)) {
+                throw new InvalidFixerConfigurationException($this->getName(), sprintf(
+                    'Expected one of "%s", got "%s".',
+                    implode('", "', array_keys(self::$tokenMap)),
+                    is_object($controlStatement) ? get_class($controlStatement) : gettype($controlStatement)
+                ));
+            }
+
+            $this->fixTokenMap[$controlStatement] = self::$tokenMap[$controlStatement];
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function fix(\SplFileInfo $file, Tokens $tokens)
+    {
+        $lineEnding = $this->whitespacesConfig->getLineEnding();
+
+        for ($index = 0, $limit = $tokens->count(); $index < $limit; ++$index) {
+            $token = $tokens[$index];
+
+            $isTokenKind = false;
+
+            foreach ($this->fixTokenMap as $tokenKind) {
+                if ($token->isGivenKind($tokenKind)) {
+                    $isTokenKind = true;
+                    break;
+                }
+            }
+
+            if (!$isTokenKind) {
+                continue;
+            }
+
+            $prevNonWhitespaceToken = $tokens[$tokens->getPrevNonWhitespace($index)];
+
+            if (!$prevNonWhitespaceToken->equalsAny(array(';', '}'))) {
+                continue;
+            }
+
+            $prevToken = $tokens[$index - 1];
+
+            if ($prevToken->isWhitespace()) {
+                $parts = explode("\n", $prevToken->getContent());
+                $countParts = count($parts);
+
+                if (1 === $countParts) {
+                    $prevToken->setContent(rtrim($prevToken->getContent(), " \t").$lineEnding.$lineEnding);
+                } elseif (count($parts) <= 2) {
+                    $prevToken->setContent($lineEnding.$prevToken->getContent());
+                }
+            } else {
+                $tokens->insertAt($index, new Token(array(T_WHITESPACE, $lineEnding.$lineEnding)));
+
+                ++$index;
+                ++$limit;
+            }
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
```

Comment:
Added a docblock here, however, in projects I work on I usually omit a docblock of this kind if

* it is overriding a method in a parent class (that has a docblock)
* implements an interface (that has a docblock)

as I believe they do not really add any value - aren't they automatically inherited?

Not sure, wasn't there a PR for adding a fixer that removes 

```php
/**
 * {@inheritdoc}
*/
```

as well?